File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.go
func maxProfit(prices []int) int {
    maxProfit := 0

    for i := 1; i < len(prices); i++ {
        if val := prices[i] - prices[i - 1]; val > 0 {
            maxProfit += val
        }
    }

    return maxProfit
}

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = INT_MAX;

        int totalProfit = 0;

        for (int i = 0; i < prices.size(); i++) {
            minPrice = min(minPrice, prices[i]);
            int profit = prices[i] - minPrice;

            if (profit > 0) {
                totalProfit += profit;
                minPrice = prices[i];        
            }
        }

        return totalProfit;
    }
};